src/components/Articles/CommentForm.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react"
+import axiosInstance from "../utilities/axios"
+
+const CommentForm = ({ articleId, onCommentSubmit }) => {
+  const [commentText, setCommentText] = useState("")
+  const [loading, setLoading] = useState(false)
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+
+    if (!commentText.trim()) {
+      alert("Please enter a comment before submitting.")
+      return
+    }
+
+    try {
+      setLoading(true)
+
+      const response = await axiosInstance.post(
+        `/articles/${articleId}/comments`,
+        {
+          username: "happyamy2016",
+          body: commentText,
+        }
+      )
+
+      // reset
+      setCommentText("")
+
+      onCommentSubmit(response.data.comment)
+
+      alert("Comment submitted successfully!")
+    } catch (error) {
+      console.error("Error submitting comment:", error)
+      alert("Failed to submit comment. Please try again later.")
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <textarea
+        value={commentText}
```

Comment:
Great work implementing this controlled component, using both `value` and `onChange` to achieve this 👍